### DISCUSSION TITLE
qbittorrent

### DISCUSSION THREAD

**Discussion by towerhand** [tteck/Proxmox#1566] [Help Needed]
Has anyone successfully migrated all legal torrents from qbittorrent in docker to the LXC created using tteckster script? I was able to find and migrate config files to ./config, but the torrents are not appearing in qbittorrent. Thanks.

**Comment by oOStroudyOo** [comment]
I believe you do something along these lines..

Go into your settings in qBittorrent and turn off automatically start downloading.
The upload the .torrent file (if you know where it is).
Then let qBittorrent do a file check for existing files before starting it.
Once you've got them readded, go back into your setting and reenable auto start downloading.

I went the other way today and I moved from Tteck LXC to Docker because the Linux Repo for qBittorrent is 2 years behind.

I had nearly 1000 torrents seeding and there was no way that I was going to manually re-add them 1 by 1 so I just cleared down my Downloads folder and started from scratch in a fresh container.

I couldn't find any mass adding solutions on the internet, and even when I set qBittorrent to monitor the Downloads folder, it still didn't pick up the .torrent files and readd them automatically.

**Reply by towerhand** [reply]
I was unaware that the qbittorrent on the Linux repo was so old. It is not worth it to migrate to it then. Thanks.

**Reply by tteck** [reply]
What advantages does it bring?

**Reply by oOStroudyOo** [reply]
If you're referring to the version of qBittorrent, I found that the older version would crash sometimes when torrents would complete, or especially when removing the torrents via the Arrs for those that were stuck on "Downloading Metadata". 
I noticed this way more when downloading music as there seems to be far less seeders for torrents.
The more torrents that was running, the more often that it would crash.
I monitored the hardware resources so I knew it wasn't a lack of RAM and etc, so I updated the version of qBit instead to see if I would get an improvement and I did.

I've not seen any crashes since moving to the latest version and I have around the same number of torrents as I did before the update, in fact it's so much more responsive, as soon as I click add or remove via the Arrs, qbittorrent deals with the torrents instantly with no delays or crashes.

I did go through all of the release notes for qBittorrent and one of the fixes sounded like it did fix the issues that I was seeing with the crashes.

**Reply by gladiatormax** [reply]
Can it be updated in the LXC from tteck?

**Reply by oOStroudyOo** [reply]
I guess you can pull it manually, or you can adjust your sources in /etc/apt/sources.list to use the SID version of Debian which will currently pull version 4.6.5 of qBittorrent.
`deb http://ftp.debian.org/debian sid main contrib`

When you run updates and upgrades it'll pull this image:
https://packages.debian.org/sid/qbittorrent-nox

I've tried it and it works.

**Reply by gladiatormax** [reply]
Hi and thx for the information, but it did not upgrade qbit for me :-(
I adjusted the file like this,
deb http://ftp.debian.org/debian sid main contrib

deb http://deb.debian.org/debian bookworm-updates main contrib

deb http://security.debian.org bookworm-security main contrib

It did a lot off updates except qbit, am I missing something?

**Reply by oOStroudyOo** [reply]
I believe qBittorrent-nox runs as a service, did you reboot it after running updates?

**Reply by gladiatormax** [reply]
yes i did reboot the container.
I did try again and look @ the upgrade list,

![image](https://github.com/tteck/Proxmox/assets/1846841/cef0c372-cf61-4db5-b21d-e88254d4eff6)

**Reply by gladiatormax** [reply]
Ok solved it by doing 'apt install qbittorrent-nox'
It upgraded and installed all dependencies.
Thx !

**Reply by oOStroudyOo** [reply]
Don't quote me on this, but `apt-get full-upgrade` might have worked for you.

**Reply by tesla-76** [reply]
Thanks, through these tips I was able to upgrade qBittorrent to the latest version as well.

I edited the **_/etc/apt/sources.list_** file by adding the string _**deb http://ftp.debian.org/debian sid main contrib**_ and restarted the LXC container. Then sudo apt-get update then sudo apt-get full-upgrade and it was updated.
